services/board/src/Domain/Pieces/Bishop.php

```diff
@@ -2,15 +2,21 @@
 
 namespace CliChess\Board\Domain\Pieces;
 
+use CliChess\Board\Domain\MovingStrategies\Diagonal;
+use CliChess\Board\Domain\MovingStrategies\Moving;
 use CliChess\Board\Domain\Square;
 
 final class Bishop implements Piece
 {
-    public function canMove(Square $from, Square $to): bool
+    private Moving $strategy;
+
+    public function __construct()
     {
-        $rowDiff = $to->rowDiff($from);
-        $columnDiff = $to->columnDiff($from);
+        $this->strategy = new Diagonal();
+    }
 
-        return abs($rowDiff) === abs($columnDiff);
+    public function canMove(Square $from, Square $to): bool
+    {
+        return $this->strategy->canMove($from, $to);
     }
```

Comment:
Now, this function can move to an abstract, wdyt?

Reply:
Yes, it makes sense.